### PR TITLE
feat(codeowners): ExternalActorSerializerBase should require integration_id

### PR DESCRIPTION
**Pull request by NisanthanNanthakumar**
## Objective:
We are transitioning towards making the `integration_id` column in `sentry_externalactor` a NOT NULL column.
This PR implements a django validation when saving records to `sentry_externalactor` and raises the appropriate errors.

There will be a follow up PR to implement the NOT NULL migration.

## Tests:
Refactored associated tests.

### CODE REVIEW

**Comment by ceorourke** [comment]
I probably missed the conversation but how does this work for codeowners in the case where they don't have an integration (the API approach)? I'm a bit out of the loop on this project so forgive me if I'm wildly misinterpreting things.

**Comment by NisanthanNanthakumar** [comment]
> I probably missed the conversation but how does this work for codeowners in the case where they don't have an integration (the API approach)? I'm a bit out of the loop on this project so forgive me if I'm wildly misinterpreting things.

@ceorourke So we are implementing a `Manual Source Code` Integration. This creates an `"dummy"` integration that doesn't link to any external service, but we can use this integration_id.

**Review by ceorourke — APPROVED** [review]

File: src/sentry/api/bases/external_actor.py [hunk 1]
```diff
@@ -27,13 +27,13 @@ class ExternalActorSerializerBase(CamelSnakeModelSerializer):  # type: ignore
...
-    integration_id = serializers.IntegerField(required=False, allow_null=True)
+    integration_id = serializers.IntegerField(required=True)
 
     @property
     def organization(self) -> Organization:
         return self.context["organization"]
 
-    def validate_integration_id(self, integration_id: str) -> Optional[str]:
+    def validate_integration_id(self, integration_id: str) -> str:
         return validate_integration_id_option(integration_id, self.organization)
```
Comment (mgaeta) [inline src/sentry/api/bases/external_actor.py:37]: `validate_integration_id_option` actually allows a null `integration_id`. You should use `validate_integration_id`.

File: src/sentry/api/validators/external_actor.py [hunk 2]
```diff
@@ -31,9 +31,7 @@ def validate_external_id_option(external_id: Optional[str]) -> Optional[str]:
 
 def validate_integration_id_option(
     integration_id: Optional[str], organization: Organization
```
Comment (mgaeta) [inline src/sentry/api/validators/external_actor.py:33]: You should create a new function `validate_integration_id` that does everything `validate_integration_id_option` does and then you can just have something like 
```
def validate_integration_id_option(...):
    if not integration_id:
        return None
	return validate_integration_id(...)
```

Comment (mgaeta) [reply src/sentry/api/validators/external_actor.py:33]: Alternatively you could just rename `validate_integration_id_option` -> `validate_integration_id` in this case.

**Pull request merged**